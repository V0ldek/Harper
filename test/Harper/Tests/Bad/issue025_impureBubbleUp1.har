ref RefBox a = {
    data = {
        a :: a;
    }

    ctor :: a -> impure -> RefBox a;
    ctor a = {
        this.data.a := a;
        return this;
    };

    get :: impure -> a;
    get = this.data.a;

    set :: a -> impure -> ();
    set a = {
        this.data.a := a;
    };
};

f :: RefBox Integer -> Integer;
f b = 42;

main :: Integer;
main = {
    (b :: RefBox Integer) = RefBox.ctor 42 ();
    return f b;
};