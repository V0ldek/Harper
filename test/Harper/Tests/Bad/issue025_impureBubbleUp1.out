Error: cannot match function's `f` declared type `RefBox Integer -> Integer` with its actual type `RefBox Integer -> impure -> Integer`.
During evaluation of:
 f b = 42
Located at line 22 column 1

Execution terminated with an error: type error.